### ISSUE TITLE
octopi can not connect to Internet!

### ISSUE THREAD

**Issue by open-src** [KaOSx/apps#149]
when install a package using octopi, an error occurs.
The error information is "Internet seems to be unavailable!", but the network is really up!

**Comment by demmm** [comment]
Post the output of doing a manual package install/using pacman:
```
sudo pacman -Syu
```
sudo pacman -S package_name
```
Use the above 2 commands & post the complete output, including commands used (thus showing what package you are trying to install.
Also post 2 files:
```
kate /etc/pacman.conf
```
```
kate /etc/pacman.d/mirrorlist
```

**Comment by open-src** [comment]
#LANG=en_GB&&sudo pacman -Syu
```
:: Synchronizing package databases...
 core is up to date
 main is up to date
 apps is up to date
:: Starting full system upgrade...
 there is nothing to do

```
 #LANG=en_GB&&sudo pacman -S dolphin-plugins
```
resolving dependencies...
looking for conflicting packages...

Package (1)           New Version  Net Change  Download Size

apps/dolphin-plugins  18.04.0-1      2.04 MiB       0.28 MiB

Total Download Size:   0.28 MiB
Total Installed Size:  2.04 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages...
 dolphin-plugins-18....   281.8 KiB   752K/s 00:00 [-------------------] 100%
(1/1) checking keys in keyring                     [-------------------] 100%
(1/1) checking package integrity                   [-------------------] 100%
(1/1) loading package files                        [-------------------] 100%
(1/1) checking for file conflicts                  [-------------------] 100%
(1/1) checking available disk space                [-------------------] 100%
:: Processing package changes...
(1/1) installing dolphin-plugins                   [-------------------] 100%
Optional dependencies for dolphin-plugins
    git: git support [installed]
    subversion: svn support
```
**The /etc/pacman.conf  file:**
```
#
# /etc/pacman.conf
#
# See the pacman.conf(5) manpage for option and repository directives

#
# GENERAL OPTIONS
#
[options]
# The following paths are commented out with their default values listed.
# If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
#CacheDir    = /var/cache/pacman/pkg/
#LogFile     = /var/log/pacman.log
#HookDir     = /etc/pacman.d/hooks/
HoldPkg     = pacman glibc
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#XferCommand = /usr/bin/curl -C - %u > %o
#CleanMethod = KeepInstalled
#UseDelta    = 0.7
Architecture = auto

# Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
#IgnorePkg   =
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

# Misc options (all disabled by default)
#UseSyslog
#ShowSize
Color
#TotalDownload
CheckSpace
VerbosePkgLists
ILoveCandy

# By default, pacman accepts packages signed by keys that its local keyring
# trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel = Never
LocalFileSigLevel = Never
#RemoteFileSigLevel = Required

#
# REPOSITORIES
#   - can be defined here or included from another file
#   - pacman will search repositories in the order defined here
#   - local/custom mirrors can be added here or in separate files
#   - repositories listed first will take precedence when packages
#     have identical names, regardless of version number
#   - URLs will have $repo replaced by the name of the current repo
#   - URLs will have $arch replaced by the name of the architecture
#
# Repository entries are of the format:
#       [repo-name]
#       Server = ServerName
#       Include = IncludePath
#
# The header [repo-name] is crucial - it must be present and
# uncommented to enable the repo.
#

[core]
Include = /etc/pacman.d/mirrorlist

[main]
Include = /etc/pacman.d/mirrorlist

[apps]
Include = /etc/pacman.d/mirrorlist

# An example of a custom package repository.  See the pacman manpage for
# tips on creating your own repositories.
#[custom]
#Server = file:///home/custompkgs

```
**The " /etc/pacman.d/mirrorlist" file:**
```
#
# KaOS repository mirrorlist
#
# Vietnam
Server = http://mirror.freedif.org/KaOS/$repo

# Czech Republic
Server = http://193.86.184.157/kaos/$repo


# Default mirror, USA
Server = https://mirror.math.princeton.edu/pub/kaoslinux/$repo

# Germany
Server = http://mirror6.layerjet.com/kaos/$repo
Server = https://mirror.alpix.eu/kaos/$repo
Server = https://kaos-frankfurt.tk/repo/$repo


# France
Server = http://kaos.youm.org/$repo
Server = http://mirror.lignux.com/kaos/$repo

# Netherlands
Server = http://nyxi.eu/mirrors/kaos/$repo

# South-Africa
Server = http://ftp.wa.co.za/pub/kaos/$repo

# United States
Server = http://mirror.dacentec.com/kaosx/$repo
Server = http://kaosx.tk/repo/$repo
Server = https://repo.kaosx.us/$repo
Server = https://kqtos.tk/repo/$repo

**Comment by open-src** [comment]
But I can't install any packages using octopi, and it complains ""Internet seems to be unavailable!"".

**Comment by demmm** [comment]
So, do you still have any errors in Octopi?  What you post now, all works as should.  Only not sure why you uncommented all mirrors.  Is Vietnam really the best location/mirror for you?

**Comment by demmm** [comment]
Run Octopi in debug mode, see if it shows then why it can't connect.  Post the full output of the following command, command included (make sure Octopi is closed first):
```
octopi -d
```
and post the output of `pacman -Qi octopi`

**Comment by open-src** [comment]
Yes, I have just updated system using pacman -Syu, and when Octopi is starting the output is: 
```
"Checking core main apps... 

Mirror-check...

'core' is synced.
'main' is synced.
'apps' is synced.

Command finished OK!
```
But when I install a package using Ocotpi, it still complains:" Internet seems to be unavailable!"

**Comment by demmm** [comment]
Please read the question, and do as asked, run in DEBUG mode and post FULL output, including command and post pacman -Qi octopi

**Comment by open-src** [comment]
**LANG=en_GB&&octopi -d
```
Time elapsed obtaining pkgs from 'ALL group' list: 473 mili seconds.
Time elapsed setting outdated foreign pkgs from 'ALL group' list: 474 mili seconds.
Time elapsed iterating over all pkgs from  'ALL group' list: 516 mili seconds.
Time elapsed setting the list to the treeview: 519 mili seconds.
2052 pkgs => Time elapsed building pkgs from 'ALL group' list: 619 mili seconds.

_print (begin):  Checking core main apps... <br>
_print (end):  Checking core main apps... <br>
_print (begin): <b>Mirror-check...</b><br><br>
_print (begin): 'core' is synced.
_print (end): 'core' is synced.
_print (begin): 'main' is synced.
_print (end): 'main' is synced.
_print (begin): 'apps' is synced. 
_print (end): 'apps' is synced.
```

**Comment by open-src** [comment]
**LANG=en_GB&&pacman**** -Qi octopi    
```
Name            : octopi
Version         : 0.8.131-1
Description     : This is Octopi, a powerful Pacman frontend using Qt libs
Architecture    : x86_64
URL             : https://octopiproject.wordpress.com/
Licenses        : GPL2
Groups          : system
Provides        : None
Depends On      : pacman  pkgfile  knotifications  alpm_octopi_utils
                  qtermwidget
Optional Deps   : None
Required By     : None
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 5.33 MiB
Packager        : Anke Boersma <demm@kaosx.us>
Build Date      : Mon May 7 17:28:56 2018
Install Date    : Sat May 12 20:56:35 2018
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : SHA-256 Sum
```

**Comment by demmm** [comment]
Did you try now to install a package too?  This output does not show you did....

**Comment by open-src** [comment]
When I select a package, for example, "firefox", and then click the "commit" button, and Ocotpi pops up a error dialog which display  "Internet seems to be unavailable!". But the console don't output anything after the line "_print (end): 'apps' is synced.".

**Comment by open-src** [comment]
I do those things just now.

**Comment by demmm** [comment]
Then please try a different mirror, see if it that changes any.  Octopi should show some clearer error in debug mode....
Edit:
```
kate /etc/pacman.d/mirrolist
```
And place a `#` in front of `Server = http://mirror.freedif.org/KaOS/$repo` and `Server = http://193.86.184.157/kaos/$repo` then save.
Use pacman one more to time to sync (forced this time:
```
sudo pacman -Syy
```
and open octopi in debug again, try installing any.

**Comment by open-src** [comment]
**sudo pacman -Syy**
```
:: Synchronizing package databases...
error: failed retrieving file 'core.db' from mirror.math.princeton.edu : Operation timed out after 10000 milliseconds with 0 out of 0 bytes received
 core                      50.0 KiB  6.42K/s 00:08 [-------------------] 100%
 main                     188.4 KiB  3.04K/s 01:02 [-------------------] 100%
error: failed retrieving file 'apps.db' from mirror.math.princeton.edu : Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds
error: failed retrieving file 'apps.db' from mirror6.layerjet.com : Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds
error: failed retrieving file 'apps.db' from mirror.alpix.eu : Operation timed out after 10000 milliseconds with 0 out of 0 bytes received
 apps                     181.8 KiB   181K/s 00:01 [-------------------] 100%
```
LANG=en_GB&&octopi -d
```
Time elapsed obtaining pkgs from 'ALL group' list: 509 mili seconds.
Time elapsed setting outdated foreign pkgs from 'ALL group' list: 510 mili seconds.
Time elapsed iterating over all pkgs from  'ALL group' list: 555 mili seconds.
Time elapsed setting the list to the treeview: 560 mili seconds.
2052 pkgs => Time elapsed building pkgs from 'ALL group' list: 674 mili seconds.

_print (begin):  Checking core main apps... <br>
_print (end):  Checking core main apps... <br>
_print (begin): <b>Mirror-check...</b><br><br>
_print (begin): 'core' is synced.
_print (end): 'core' is synced.
_print (begin): 'main' is synced.
_print (end): 'main' is synced.
_print (begin): 'apps' is not synced.
_print (end): 'apps' is not synced.
_print (begin): Do not update, wait for your mirror to sync. 
_print (end): Do not update, wait for your mirror to sync. 

```
So I select the "core" Repository, and select a package named "apr", then click the commit button, and Ocotpi still pops up a error dialog with error information "Internet seems to be unavailable!".
The console continues to output:
```
55 pkgs => Time elapsed building pkgs from 'base group' list: 159 mili seconds.

16 pkgs => Time elapsed building pkgs from 'base-devel group' list: 82 mili seconds.

Time elapsed obtaining pkgs from 'ALL group' list: 111 mili seconds.
Time elapsed setting outdated foreign pkgs from 'ALL group' list: 112 mili seconds.
Time elapsed obtaining outdated foreign pkgs from 'ALL group' list: 145 mili seconds.
Time elapsed iterating over all pkgs from  'ALL group' list: 187 mili seconds.
Time elapsed setting the list to the treeview: 192 mili seconds.
2052 pkgs => Time elapsed building pkgs from 'ALL group' list: 287 mili seconds.
```

**Comment by open-src** [comment]
The default mirror server is too slow.

**Comment by open-src** [comment]
The console still can't output available information for install a package using Octopi.
But I can install any packages using pacman -S packagename.

**Comment by open-src** [comment]
En,I click theTools->SysInfo, But Octopi still says  "Internet seems to be unavailable!".
And I used KaOS last year, Octopi works normally. 
I only use pacman to manage packages in console now.

**Comment by open-src** [comment]
The issue occurs in my  two computers.

**Comment by demmm** [comment]
Please read questions in full......
>Can you replicate in Live mode?

Since no-one can reproduce, there is something you altered on your systems, so please respond if it works on a clean system, thus use the ISO KaOS 2018.04 to check.

**Comment by open-src** [comment]
In Live mode , the SysInfo tool can works in Octopi, the output information is following:
```
----------------------------------------------------------------------------------------------------------
cat /etc/KaOS-release
----------------------------------------------------------------------------------------------------------



----------------------------------------------------------------------------------------------------------
inxi -Fxz
----------------------------------------------------------------------------------------------------------

System:    Host: <HOSTNAME> Kernel: 4.15.7-1 x86_64 bits: 64 compiler: gcc v: 7.3.0 
           Desktop: KDE Plasma 5.12.4 tk: Qt 5.10.1 Distro: KaOS (2018) 
Machine:   Type: Desktop Mobo: Intel model: NUC6i5SYB v: H81131-503 serial: N/A 
           UEFI [Legacy]: Intel v: SYSKLi35.86A.0061.2017.0524.1454 date: 05/24/2017 
CPU:       Topology: Dual Core model: Intel Core i5-6260U bits: 64 type: MT MCP arch: Skylake 
           rev: 3 L2 cache: 4096 KiB 
           flags: lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 14400 
           Speed: 500 MHz min/max: 400/2900 MHz Core speeds (MHz): 1: 500 2: 500 3: 500 4: 500 
Graphics:  Card-1: Intel Iris Graphics 540 driver: i915 v: kernel bus ID: 00:02.0 
           Display: x11 server: X.Org 1.19.6 driver: intel unloaded: fbdev,modesetting,vesa 
           resolution: 1920x1200~60Hz 
           OpenGL: renderer: Mesa DRI Intel Iris Graphics 540 (Skylake GT3e) v: 4.5 Mesa 17.3.9 
           direct render: Yes 
Audio:     Card-1: Intel Sunrise Point-LP HD Audio driver: snd_hda_intel v: kernel bus ID: 00:1f.3 
           Sound Server: ALSA v: k4.15.7-1 
Network:   Card-1: Intel Ethernet Connection I219-V driver: e1000e v: 3.2.6-k port: N/A 
           bus ID: 00:1f.6 
           IF: eno1 state: down mac: <filter> 
           Card-2: Intel Wireless 8260 driver: iwlwifi v: kernel bus ID: 01:00.0 
           IF: wlp1s0 state: up mac: <filter> 
           Card-3: Ralink RT5370 Wireless Adapter type: USB driver: rt2800usb bus ID: 1:3 
           IF: wlp0s20f0u2 state: down mac: <filter> 
Drives:    HDD Total Size: 2.04 TiB used: 1.86 GiB (0.1%) 
           ID-1: /dev/sda model: ST2000LM007-1R81 size: 1.82 TiB 
           ID-2: /dev/sdb model: INTEL_SSDSCKJW24 size: 223.57 GiB 
           ID-3: /dev/sdc type: USB model: Cruzer_Blade size: 3.75 GiB 
Partition: ID-1: / size: 11.63 GiB used: 76.1 MiB (0.6%) fs: aufs dev: ERR-102 
Sensors:   System Temperatures: cpu: 52.0 C mobo: 27.8 C 
           Fan Speeds (RPM): N/A 
Info:      Processes: 161 Uptime: 20m Memory: 15.50 GiB used: 868.1 MiB (5.5%) Init: systemd 
           runlevel: 5 Compilers: gcc: 7.3.0 Client: Unknown Client: octopi inxi: 3.0.06 


----------------------------------------------------------------------------------------------------------
journalctl -b -p err
----------------------------------------------------------------------------------------------------------



----------------------------------------------------------------------------------------------------------
cat /etc/pacman.conf
----------------------------------------------------------------------------------------------------------

#
# /etc/pacman.conf
#
# See the pacman.conf(5) manpage for option and repository directives

#
# GENERAL OPTIONS
#
[options]
# The following paths are commented out with their default values listed.
# If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
#CacheDir    = /var/cache/pacman/pkg/
#LogFile     = /var/log/pacman.log
HoldPkg     = pacman glibc pacman-helpme
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#XferCommand = /usr/bin/curl %u > %o
#CleanMethod = KeepInstalled
#UseDelta    = 0.7
Architecture = auto

# Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
IgnorePkg   = 
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

# Misc options (all disabled by default)
#NoPassiveFtp
#UseSyslog
Color
#TotalDownload
CheckSpace
VerbosePkgLists
ILoveCandy

# By default, pacman accepts packages signed by keys that its local keyring
# trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel = Never
LocalFileSigLevel = Never
#RemoteFileSigLevel = Required

#
# REPOSITORIES
#   - can be defined here or included from another file
#   - pacman will search repositories in the order defined here
#   - local/custom mirrors can be added here or in separate files
#   - repositories listed first will take precedence when packages
#     have identical names, regardless of version number
#   - URLs will have $repo replaced by the name of the current repo
#
# Repository entries are of the format:
#       [repo-name]
#       Server = ServerName
#       Include = IncludePath
#
# The header [repo-name] is crucial - it must be present and
# uncommented to enable the repo.
#

[core]
Include = /etc/pacman.d/mirrorlist

[main]
Include = /etc/pacman.d/mirrorlist

[apps]
Include = /etc/pacman.d/mirrorlist

# An example of a custom package repository.  See the pacman manpage for
# tips on creating your own repositories.
#[custom]
#Server = file:///home/custompkgs


----------------------------------------------------------------------------------------------------------
pacman -Qm
----------------------------------------------------------------------------------------------------------

libldap 2.4.46-2


----------------------------------------------------------------------------------------------------------
cat /var/log/pacman.log
----------------------------------------------------------------------------------------------------------



----------------------------------------------------------------------------------------------------------
cat /var/log/installation.log
----------------------------------------------------------------------------------------------------------


```

**Comment by demmm** [comment]
How do you connect to the internet?  Since it can't be replicated and clearly not a user setting change, it could be that your ISP/router has some setting that trips up Octopi.  I'll probably will need upstream Octopi dev to help here and figure why an ISP setting can fail Octopi.

**Comment by open-src** [comment]
I recompile and install octopi using PKGBUILD in archlinux(octopi-0.8.5-3-x86_64.pkg.tar.xz), it runs normally.
The PKGBUILD is:
```
# Maintainer: Jameson Pugh <imntreal@gmail.com>

pkgbase=octopi
pkgname=('octopi' 'octopi-pacmanhelper' 'octopi-notifier-qt5' 'octopi-notifier-frameworks' 'octopi-repoeditor' 'octopi-cachecleaner')
pkgver=0.8.5
pkgrel=3
# This is the release package so the below _gitcommit variable should (usually) be commented out.
_gitcommit="52e111f5ab53b5b60ac06ca1cd47c70e0aced435"
pkgdesc="a powerful Pacman frontend using Qt libs"
arch=('i686' 'x86_64')
url="http://octopiproject.wordpress.com"
license=('GPL2')
makedepends=('qt5-declarative' 'git' 'knotifications' 'alpm_octopi_utils' 'qtermwidget')
if [ "${_gitcommit}" != "" ]; then
	source=("octopi-${pkgver}-${pkgrel}.tar.gz::https://github.com/aarnt/octopi/archive/${_gitcommit}.tar.gz"
	'octopi-repoeditor.desktop')
else
  source=("https://github.com/aarnt/${pkgname}/archive/v${pkgver}.tar.gz"
	'octopi-repoeditor.desktop')
fi
sha256sums=('fda4b2c532d8f398ea03bfaf44fa86ab3a1d9611e652b313807a3193ad5a7e56'
            '131f16745df685430db55e54ede6da66aed9b02ca00d6d873a002b2a3e1c90ef')

prepare() {
  if [ "${_gitcommit}" != "" ]; then
    cd "${srcdir}/${pkgbase}-${_gitcommit}"
  else
	  cd "${srcdir}/${pkgbase}-${pkgver}"
  fi
  
	# sed version
  sed -i -e "s|0.9.0 (dev)|${pkgver}-${pkgrel}|g" src/strconstants.h

  cp -r notifier notifier-qt5
  cp -r notifier notifier-frameworks
	sed -i 's|#DEFINES += KSTATUS|DEFINES += KSTATUS|' notifier-frameworks/octopi-notifier/octopi-notifier.pro
}            
           
build() {
	if [ "${_gitcommit}" != "" ]; then
    cd "${srcdir}/${pkgbase}-${_gitcommit}"
	else
  	cd "${srcdir}/${pkgbase}-${pkgver}"
	fi

	msg "Building octopi..."
	qmake-qt5 octopi.pro
	make

	cd notifier/pacmanhelper
	msg "Building pacmanhelper..."
	qmake-qt5 pacmanhelper.pro
	make

	cd ../../notifier-qt5/octopi-notifier
	msg "Building octopi-notifier-qt5..."
	qmake-qt5 octopi-notifier.pro
	make
  
	cd ../../notifier-frameworks/octopi-notifier
	msg "Building octopi-notifier-frameworks..."
	qmake-qt5 octopi-notifier.pro
	make
  
	cd ../../repoeditor
	msg "Building octopi-repoeditor..."
	qmake-qt5 octopi-repoeditor.pro
	make

	cd ../cachecleaner
	msg "Building octopi-cachecleaner..."
	qmake-qt5 octopi-cachecleaner.pro
	make
}

package_octopi() {
	pkgdesc="A powerful Pacman frontend using Qt5 libs"
	depends=('qt5-declarative' 'alpm_octopi_utils' 'qtermwidget')
	optdepends=('xterm: for AUR support'
				'kdesu: for KDE'
		    'gksu: for XFCE, Gnome, LXDE, Cinnamon'
				'lxqt-sudo: for LXQT'
		    'gnome-keyring: for password management'
		    'gist: for SysInfo report'
		    'pacaur: for AUR support'
		    'yaourt: for AUR support'
		    'octopi-repoeditor: for editing functions'
		    'octopi-cachecleaner: for cleaning functions'
		    'octopi-notifier-qt5: for notifications'
		    'octopi-notifier-frameworks: for notifications'
		    'pacmanlogviewer: to view pacman log files')
	conflicts=('octopi-git')

	if [ "${_gitcommit}" != "" ]; then
    cd "${srcdir}/${pkgbase}-${_gitcommit}"
	else
		cd "${srcdir}/${pkgbase}-${pkgver}"
	fi
 
	install -D -m755 "bin/${pkgname}" "${pkgdir}/usr/bin/${pkgname}"
	install -D -m644 "${pkgname}.desktop" "${pkgdir}/usr/share/applications/${pkgname}.desktop"
	install -D -m644 "resources/images/${pkgname}_green.png" "${pkgdir}/usr/share/icons/${pkgname}.png"
	install -D -m644 "resources/images/${pkgname}_green.png" "${pkgdir}/usr/share/icons/gnome/32x32/apps/${pkgname}.png"
	install -D -m644 "resources/images/${pkgname}_red.png" "${pkgdir}/usr/share/icons/${pkgname}_red.png"
	install -D -m644 "resources/images/${pkgname}_yellow.png" "${pkgdir}/usr/share/icons/${pkgname}_yellow.png"
  
	#speedup files
	install -D -m755 "speedup/speedup-octopi.sh" "${pkgdir}/usr/bin/speedup-octopi.sh"
	install -D -m644 "speedup/${pkgname}.service" "${pkgdir}/usr/lib/systemd/system/${pkgname}.service"
}

package_octopi-pacmanhelper() {
	pkgdesc="Pacman helper for Octopi notifier"
  depends=('qt5-base')
	
  if [ "${_gitcommit}" != "" ]; then
    cd "${srcdir}/${pkgbase}-${_gitcommit}"
  else
    cd "${srcdir}/${pkgbase}-${pkgver}"
  fi
	
  install -Dm755 "notifier/bin/pacmanhelper" "${pkgdir}/usr/lib/octopi/pacmanhelper"
	install -Dm644 "notifier/pacmanhelper/polkit/org.octopi.pacman.policy" "${pkgdir}/usr/share/polkit-1/actions/org.octopi.pacman.policy"
	install -Dm644 "notifier/pacmanhelper/polkit/org.octopi.pacmanhelper.conf" "${pkgdir}/etc/dbus-1/system.d/org.octopi.pacmanhelper.conf"
	install -Dm644 "notifier/pacmanhelper/polkit/org.octopi.pacmanhelper.xml" "${pkgdir}/usr/share/dbus-1/interfaces/org.octopi.pacmanhelper.xml"
	install -Dm644 "notifier/pacmanhelper/polkit/org.octopi.pacmanhelper.service" "${pkgdir}/usr/share/dbus-1/system-services/org.octopi.pacmanhelper.service"
}

package_octopi-notifier-qt5() {
	pkgdesc="Notifier for Octopi using Qt5 libs"
	depends=('octopi-pacmanhelper' 'libnotify' 'qt5-base')
	optdepends=('xfce4-notifyd: for notifications in XFCE')
	conflicts=('octopi-notifier' 'octopi-notifier-qt4' 'octopi-notifier-frameworks')
	provides=('octopi-notifier')
	replaces=('octopi-notifier-qt4')
	
	if [ "${_gitcommit}" != "" ]; then
		cd "${srcdir}/${pkgbase}-${_gitcommit}"
	else
		cd "${srcdir}/${pkgbase}-${pkgver}"
	fi
	
	#Octopi-notifier files
	install -D -m755 "notifier-qt5/bin/octopi-notifier" "${pkgdir}/usr/bin/octopi-notifier"
	install -D -m644 "notifier-qt5/octopi-notifier/octopi-notifier.desktop" "${pkgdir}/usr/share/applications/${pkgname}.desktop"
	install -D -m644 "notifier-qt5/octopi-notifier/octopi-notifier.desktop" "${pkgdir}/etc/xdg/autostart/${pkgname}.desktop"
}

package_octopi-notifier-frameworks() {
	pkgdesc="Notifier for Octopi with Knotifications support"
	depends=('octopi-pacmanhelper' 'libnotify' 'knotifications')
	optdepends=('xfce4-notifyd: for notifications in XFCE')
	conflicts=('octopi-notifier' 'octopi-notifier-qt4' 'octopi-notifier-qt5')
	provides=('octopi-notifier')
	replaces=('octopi-notifier')
	
	if [ "${_gitcommit}" != "" ]; then
		cd "${srcdir}/${pkgbase}-${_gitcommit}"
	else
		cd "${srcdir}/${pkgbase}-${pkgver}"
	fi
	
	#Octopi-notifier files
	install -D -m755 "notifier-frameworks/bin/octopi-notifier" "${pkgdir}/usr/bin/octopi-notifier"
	install -D -m644 "notifier-frameworks/octopi-notifier/octopi-notifier.desktop" "${pkgdir}/usr/share/applications/${pkgname}.desktop"
	install -D -m644 "notifier-frameworks/octopi-notifier/octopi-notifier.desktop" "${pkgdir}/etc/xdg/autostart/${pkgname}.desktop"
}

package_octopi-repoeditor() {
	pkgdesc="Pacman repo editor for Octopi"
	depends=('qt5-base')
	
	if [ "${_gitcommit}" != "" ]; then
  	cd "${srcdir}/${pkgbase}-${_gitcommit}"
	else
	  cd "${srcdir}/${pkgbase}-${pkgver}"
	fi
	
	#Octopi-repoeditor files
	install -D -m755 "repoeditor/bin/octopi-repoeditor" "${pkgdir}/usr/bin/octopi-repoeditor"
	install -D -m644 "${srcdir}/${pkgname}.desktop" "${pkgdir}/usr/share/applications/${pkgname}.desktop"
	install -D -m644 "resources/images/${pkgbase}_red.png" "${pkgdir}/usr/share/icons/${pkgname}.png"
}

package_octopi-cachecleaner() {
	pkgdesc="Cachecleaner for Octopi"
	depends=('qt5-base')
	
	if [ "${_gitcommit}" != "" ]; then
	  cd "${srcdir}/${pkgbase}-${_gitcommit}"
	else
		cd "${srcdir}/${pkgbase}-${pkgver}"
	fi
	
	#Octopi-cachecleaner files
	install -D -m755 "cachecleaner/bin/octopi-cachecleaner" "${pkgdir}/usr/bin/octopi-cachecleaner"
	install -D -m644 "cachecleaner/$pkgname.desktop" "${pkgdir}/usr/share/applications/$pkgname.desktop"
}

# vim: set ts=2 sw=2 ft=sh noet:

```

**Comment by open-src** [comment]
I recommend that octopi should update to this version:"octopi-0.8.5-3-x86_64.pkg.tar.xz".  ^_^

**Comment by demmm** [comment]
KaOS uses a much newer version then you show above, that one is fully broken for KaOS.  The version in KaOS is based on the very latest commits.
KaOS is not Arch or Arch based, and uses very different tools in Octopi (kcp, SysInfo for example).  
But if one of later commits causes your network issue, then this is an upstream change.
See commit from 11 days ago:
https://github.com/KaOSx/apps/tree/master/octopi
See latest commits:
https://github.com/aarnt/octopi/commits/master

**Comment by demmm** [comment]
If you truly want to compare, build it with your above PKGBUILD in Archlinux from commit c413caadd11883ffcebf13ce515fb17bc23ebd27

**Comment by open-src** [comment]
En, I have tried it, it really an upstream issue, So this is not a bug about KaOS.
BTW, will the octopi for KaOS downgrade for the issue?

**Comment by open-src** [comment]
Thank you for your patient explanation extremely! So the issue should be closed.

**Comment by demmm** [comment]
No, since there are major changes in SysInfo, which is completely broken in older versions.
See:
https://github.com/aarnt/octopi/issues/322
I'll report this issue to aarnt (octopi dev), he'll fix it soon.  Plus, seems so far no-one else is effected by this but you, so downgrading makes double no sense.

**Comment by demmm** [comment]
Leave it open until it is fixed upstream.

**Comment by demmm** [comment]
To help upstream with this bug, it should be in one of two commits.
Please test a rebuild of Octopi with one of these commits reverted:
```
sudo pacman -U http://kaosx.tk/repo/build/octopi-0.8.131-2-x86_64.pkg.tar.xz
```
and report back if that fixes your issue.  If not, another build will be done with the other commit reverted.

**Comment by open-src** [comment]
I hava installed the version "0.8.131-2", and the issue is still not fixed.

**Comment by demmm** [comment]
One more commit reverted:
```
sudo pacman -U http://kaosx.tk/repo/build/octopi-0.8.131-3-x86_64.pkg.tar.xz
```
that leaves one more commit to test, if still not fixed.

**Comment by open-src** [comment]
I have tried this commit, and it is still not fixed either.

**Comment by open-src** [comment]
I think you can using "binary test" between the latest version and 0.8.5-3 version.

**Comment by demmm** [comment]
Please see previous comments about Octopi being broken with the commit you keep referring too.  I will revert one more commit that changed things in Octopi for testing internet connection.

**Comment by demmm** [comment]
Do you happen to use a Chinese ISP?  Looks like this might be your issue:
https://github.com/aarnt/octopi/issues/334

**Comment by demmm** [comment]
Since no further response from submitter, it is assumed this issue is fixed with Octopi 0.9.0.  Closing.
